### ISSUE TITLE
Error message is lost when publishing invalid events

### ISSUE THREAD

**Issue by GJL** [zalando-incubator/nakadi-java#116]
Nakadi returns an array instead of a single object when schema violations are found. Find example below.

`[{"publishing_status":"failed","detail":"#: 2 schema violations found\n#: ... }]`

However, `OkHttpResource.handleError()` expects a single `Problem` object:
```
Problem problem = (Problem)Optional.ofNullable(this.jsonSupport.fromJson(raw, Problem.class)) ...
```

Hence, the deserialization with gson fails with the following exception:
```
Caused by: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was BEGIN_ARRAY at line 1 column 2 path $
	at nakadi.shadow.com.google.gson.stream.JsonReader.beginObject(JsonReader.java:385)
	at nakadi.shadow.com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:213)
	... 49 more
```

**Comment by dehora** [comment]
@GJL This is an issue with the Nakadi implementation; it doesn't match its API. I'd suggest filing a bug upstream in Nakadi. Will leave this open for the time being.

**Comment by GJL** [comment]
@dehora Sorry, I am on vacation and only now had time to reply.

https://github.com/zalando/nakadi/blob/master/api/nakadi-event-bus-api.yaml#L377
```
 '422':
          description: |
            At least one event failed to be validated, enriched or partitioned. None were submitted.
          schema:
            type: array
            items:
              $ref: '#/definitions/BatchItemResponse'
```

To me it seems that the behavior is documented correctly in the Nakadi project.

**Comment by dehora** [comment]
> To me it seems that the behavior is documented correctly in the Nakadi project: The return type can be an array.

Got it. The server's also returned the wrong format for some errors, which distracted me. Marking this as  a bug in batch handling.

**Comment by dehora** [comment]
There's a patch for this in #122 which has been released as 0.7.5

**Comment by dehora** [comment]
Had a report from a customer it's working for them in 0.7.5, closing
